src/Aspire.Dashboard/Model/ResourceOutgoingPeerResolver.cs

```diff
@@ -132,17 +132,43 @@ bool TryMatchResourceAddress(string value, [NotNullWhen(true)] out string? name,
         {
             foreach (var (resourceName, resource) in resources)
             {
+                // Try to match against URL endpoints
                 foreach (var service in resource.Urls)
                 {
                     var hostAndPort = service.Url.GetComponents(UriComponents.HostAndPort, UriFormat.UriEscaped);
 
-                    if (string.Equals(hostAndPort, value, StringComparison.OrdinalIgnoreCase))
+                    if (DoesAddressMatch(hostAndPort, value))
                     {
                         name = ResourceViewModel.GetResourceName(resource, resources);
                         resourceMatch = resource;
                         return true;
                     }
                 }
+
+                // Try to match against connection strings using comprehensive parsing
+                if (resource.Properties.TryGetValue(KnownProperties.Resource.ConnectionString, out var connectionStringProperty) &&
+                    connectionStringProperty.Value.TryConvertToString(out var connectionString) &&
+                    ConnectionStringParser.TryDetectHostAndPort(connectionString, out var host, out var port))
+                {
+                    var endpoint = port.HasValue ? $"{host}:{port.Value}" : host;
+                    if (DoesAddressMatch(endpoint, value))
+                    {
+                        name = ResourceViewModel.GetResourceName(resource, resources);
+                        resourceMatch = resource;
+                        return true;
+                    }
+                }
+
+                // Try to match against parameter values (for Parameter resources that contain URLs)
```

Comment:
Use ConnectionStringParser for Parameter resources too as it is a superset of TryParseUrlHostAndPort